### DISCUSSION TITLE
vitejs not copile or purge tailwindcss in dynamics components vue

### DISCUSSION THREAD

**Discussion by mainhurt** [tailwindlabs/tailwindcss#10483] [Help] [answered]
The Vitejs config + tailwindcss.config.js a pesar que la configuracion recomendada por la página tailwindcss y vitejs, para laravel vue y postcss autoprefixer, este no procesa componentes vuejs con clases dinámicas.

resulta que en la instalación en limpio de laravel, este genera la configuración automática

Despite the configuration recommended by the tailwindcss and vitejs page, for laravel vue and postcss autoprefixer, it does not process vuejs components with dynamic classes.

It turns out that in the clean installation of Laravel, it generates the automatic configuration

**vite.config.js** :

```
export default defineConfig({
    plugins: [
        laravel({
            input: 'resources/js/app.js',
            refresh: true,
        }),
        vue({
            template: {
                transformAssetUrls: {
                    base: null,
                    includeAbsolute: false,
                },
            },
        })
    ],
});
```

**What version of Tailwind CSS are you using?**

"tailwindcss": "^3.2.4",

**What build tool (or framework if it abstracts the build tool) are you using?**

"@vitejs/plugin-vue": "^4.0.0",

"php": "^8.0.2",
Laravel Version ............................................................................. 9.50.2
 PHP Version .................................................................................. 8.1.6
 Composer Version ............................................................................. 2.5.1

**What version of Node.js are you using?**

 node --version   v18.13.0

**What browser are you using?**

opera gx

**What operating system are you using?**

Windows 11

use manual instalation:
https://github.com/laravel/vite-plugin/blob/main/UPGRADE.md#migrating-from-laravel-mix-to-vite

and the following modifications:
y las modificaciones seguientes:

**vite.config.js**

```
export default defineConfig({
    plugins: [
        laravel([
            'resources/css/app.css',
            'resources/js/app.js'
        ],{
            refresh: true,
        }),
        vue({
            template: {
                transformAssetUrls: {
                    base: null,
                    includeAbsolute: false,
                },
            },
        })
    ],
    resolve: {
        alias: {
            '@': '/resources/js',
        },
    },
});
```

**app.blade.php**

```
<!DOCTYPE html>
<html lang="{{ str_replace('_', '-', app()->getLocale()) }}">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">

        <title inertia>{{ config('app.name', 'Laravel') }}</title>

        <!-- Fonts -->
        <link rel="stylesheet" href="https://fonts.bunny.net/css2?family=Nunito:wght@400;600;700&display=swap">

        <!-- Scripts -->
        @routes
        @vite(['resources/js/app.js', "resources/js/Pages/{$page['component']}.vue"])
        @vite(['resources/css/app.css', 'resources/js/app.js'])
        @inertiaHead
    </head>
    <body class="font-sans antialiased">
        @inertia
    </body>
</html>

```

**Describe your issue**
```
let extbox = ref(['red', 'indigo', 'purple', 'blue', 'yellow', 'orange'])

<ul class="flex flex-row flex-wrap text-sm max-w-xs ">
  <li :class="`hover:bg-${color}-800 hover:text-white bg-${color}-100`" class=" w-14 h-14 m-1  border-gray-400 relative flex items-center justify-center border-2 cursor-pointer rounded-md" v-for="color, index in extbox" :key="index">
    {{ color }}
  </li>
</ul>
```

it perfectly builds the Tailwind elements but doesn't render the css inside the build in the public folder. I need help on this issue
in short it does not generate the dynamically created elements

se crea perfectamente los elementos de Tailwind pero no renderiza el CSS dentro de la compilación en la carpeta pública. requiero apoyo en este tema
en resumidas no genera los elementos dinámicamente creados

**Comment by wongjn** [comment, marked as answer]
As per [the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names):
> The most important implication of how Tailwind extracts class names is that it will only find classes that exist as complete unbroken strings in your source files.
>
> If you use string interpolation or concatenate partial class names together, Tailwind will not find them and therefore will not generate the corresponding CSS

**Reply by mainhurt** [reply]
If you are correct, I did not go through the documentation, and I see that I cannot generate dynamic components. It had already happened to me, 😂🤣 and I did it again. thanks for the info. I wasted about two days thinking it was vitejs and tried with laravel mix, postcss was giving me a stone. hahahahaha


I need create one component per color 😒

**Comment by mainhurt** [comment]
```
<script setup>
import { reactive } from 'vue';

defineProps({
  color: {
    typeof: String
  }
})
const colorVariants = reactive({
  blue: 'bg-blue-600 hover:bg-blue-800 border-blue-400',
  red: 'bg-red-600 hover:bg-red-800 border-red-400',
  indigo: 'bg-indigo-600 hover:bg-indigo-800  border-indigo-400',
  purple: 'bg-purple-600 hover:bg-purple-800 border-purple-400',
  cyan: 'bg-blue-600 hover:bg-blue-800 border-blue-400',
  yellow: 'bg-yellow-600 hover:bg-yellow-800 border-yellow-400',
  orange: 'bg-orange-600 hover:bg-orange-800 border-orange-400',
})
</script>
```
template

```
<template>
  <div>
    <ul class="flex flex-row flex-wrap text-sm max-w-xs ">
        <li class=" w-14 h-14 m-1  relative flex items-center justify-center border-2 cursor-pointer rounded-md text-white hover:font-bold capitalize"
          :class="`${colorVariants[color]}`">
          {{ color }}
        </li>
    </ul>
  </div>
</template>
```
